pkg/diagnose/connectivity/core_endpoint.go

```diff
@@ -25,15 +25,27 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config/utils"
 	"github.com/DataDog/datadog-agent/pkg/diagnose/diagnosis"
 	logshttp "github.com/DataDog/datadog-agent/pkg/logs/client/http"
+	logstcp "github.com/DataDog/datadog-agent/pkg/logs/client/tcp"
 	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
 )
 
-func getLogsHTTPEndpoints() (*logsConfig.Endpoints, error) {
+func getLogsEndpoints(useTCP bool) (*logsConfig.Endpoints, error) {
 	datadogConfig := pkgconfigsetup.Datadog()
 	logsConfigKey := logsConfig.NewLogsConfigKeys("logs_config.", datadogConfig)
+	if useTCP {
+		return logsConfig.BuildEndpointsWithConfig(datadogConfig, logsConfigKey, "agent-http-intake.logs.", false, "logs", logsConfig.AgentJSONIntakeProtocol, logsConfig.DefaultIntakeOrigin)
+	}
 	return logsConfig.BuildHTTPEndpointsWithConfig(datadogConfig, logsConfigKey, "agent-http-intake.logs.", "logs", logsConfig.AgentJSONIntakeProtocol, logsConfig.DefaultIntakeOrigin)
```

Comment:
```suggestion
	if useTCP {
		return logsConfig.BuildEndpointsWithConfig(datadogConfig, logsConfigKey, "agent-http-intake.logs.", false, "logs", logsConfig.AgentJSONIntakeProtocol, logsConfig.DefaultIntakeOrigin)
	} else {
	  return logsConfig.BuildHTTPEndpointsWithConfig(datadogConfig, logsConfigKey, "agent-http-intake.logs.", "logs", logsConfig.AgentJSONIntakeProtocol, logsConfig.DefaultIntakeOrigin)
	}
```

This is somewhat of a style preference, but I think the "early return" pattern makes this harder to read. I tend to reserve the "early return" pattern for checking preconditions and handling errors.